internal/expr/prepare.go

```diff
@@ -41,7 +79,99 @@ func prepareInput(ti typeNameToInfo, p *inputPart) error {
 	return nil
 }
 
-// Prepare takes a parsed expression and all the Go objects mentioned in it.
+// prepareOutput checks that the output expressions are correspond to a known types.
+// It then checks they are formatted correctly and finally generates the columns for the query.
+func prepareOutput(ti typeNameToInfo, p *outputPart) ([]fullName, []outputDest, error) {
+
+	var outCols = make([]fullName, 0)
+	var outDests = make([]outputDest, 0)
+
+	// Check the asterisk are well formed (if present).
+	if err := starCheckOutput(p); err != nil {
+		return nil, nil, err
+	}
+
+	// Check target struct type and its tags are valid.
+	var info *info
+	var ok bool
+
+	for _, t := range p.target {
+		info, ok = ti[t.prefix]
+		if !ok {
+			return nil, nil, fmt.Errorf(`type %s unknown, have: %s`, t.prefix, strings.Join(getKeys(ti), ", "))
+		}
+
+		if t.name != "*" {
+			f, ok := info.tagToField[t.name]
+			if !ok {
+				return nil, nil, fmt.Errorf(`type %s has no %q db tag`, info.structType.Name(), t.name)
+			}
+			// For a none star expression we record output destinations here.
+			// For a star expression we fill out the destinations as we generate the columns.
+			outDests = append(outDests, outputDest{info.structType, f})
+
+		}
```

Comment:
Nitpick: extra spaces before closings are generally not visually helpful.

Reply:
Changed.